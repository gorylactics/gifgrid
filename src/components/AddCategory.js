import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ({setCategory})=>{

    const [inputValue , setInputValue] = useState('')
    const handleInputChange = (e) =>{
        setInputValue(e.target.value)
    }; //esta const recibe el valor que se ingresa en el input

    const handleSubmit = (e)=>{
        e.preventDefault();
        if(inputValue.trim().length > 2){
            setCategory(cat => [...cat , inputValue] );
            setInputValue('')
        }
    }
    return(
        <form onSubmit={handleSubmit}>
            <input type="text" value={inputValue} onChange={handleInputChange} />
        </form>
    )
}
AddCategory.propTypes = {
    setCategory  : PropTypes.func.isRequired
}
