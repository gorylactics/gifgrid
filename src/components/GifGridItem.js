import React from 'react'

// export const GifGridItem = (gif) => {
//     console.log(gif)
//     return (
//         <div>
//             {gif.title}
//         </div>
//     )
// }

export const GifGridItem = ({id , title ,  url}) => {
    console.log(id , title ,  url)
    return (
        <div className=''>
            <h3>{title}</h3>
            <img src={url} alt={title}/>
            
        </div>
    )
}