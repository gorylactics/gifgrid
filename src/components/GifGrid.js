import React , {useEffect , useState} from "react";
import { GifGridItem } from "./GifGridItem";


export const GifGrid = ({category})=>{  
    
    const [images, setImages] = useState([])

    useEffect(()=>{
        getGifts();
    }, [])

    const getGifts = async()=>{
        const url = 'https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&limit=10&api_key=oiHbqpSMwYCqJNuEwrl9AthJWulF1Uto';
        const resp = await fetch(url);
        const {data} = await resp.json()
        // propiedad data de la resp de la api
        // const data = await resp.json() ese es el nombre que se le asigna al dato que responde el await
        const gifs = data.map(i =>{
            return{
                id: i.id,
                title : i.title,
                url : i.images?.downsized_medium.url
            }
        })
        console.log(gifs)
        setImages(gifs);
    } 
    // getGifts();
    return(
        // <>
        // <h3>{category}</h3>
        // <ol>
        //     {/* {images.map(i => <li key={i.id}>{i.title}</li>)} */}
        //     {images.map(({id , title}) => <li key={id}>{title}</li>)}
        //     {/* aca se desestructua la imagen mediante sus atributos */}
        // </ol>
        // </>
        <>
            <h3>{category}</h3>
            <div className="card-grid">
                {images.map(img =>(
                    <GifGridItem 
                        key={img.id} 
                        img={img}
                        url={img.url}
                        title={img.title}
                    />
                ))}
            </div>
        </>
    )
}