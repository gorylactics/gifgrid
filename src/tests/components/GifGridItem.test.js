import react from "react"
import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

describe('prueba en <GifGridItem />', () => {
    test('debe mostrar el componente correctamente', () => {
        const wrapped = shallow(<GifGridItem />);
        expect(wrapped).toMatchSnapshot();
    })
})
