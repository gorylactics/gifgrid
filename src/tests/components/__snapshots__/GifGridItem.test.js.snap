// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba en <GifGridItem /> debe mostrar el componente correctamente 1`] = `
<div
  className="animate__animated animate__fadeIn"
>
  <h3 />
  <img />
</div>
`;
